nntrainer/src/fc_layer_op.cpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (C) 2020 Samsung Electronics Co., Ltd. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ * @file	fc_layer.cpp
+ * @date	14 May 2020
+ * @brief	This is Fully Connected Layer Class for Neural Network
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ */
+
+#include <fc_layer.h>
+#include <layer.h>
+#include <lazy_tensor.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <parse_util.h>
+#include <util_func.h>
+
+namespace nntrainer {
+
+int FullyConnectedLayer::initialize(bool last, TensorParam input) {
+  int status = ML_ERROR_NONE;
+
+  this->last_layer = last;
+
+  output_dim = input_dim;
+  output_dim.width(unit);
+
+  Tensor out(output_dim);
+  TensorParam output = TensorParam(out, Tensor(), "FC:output");
+
+  Tensor bias = Tensor(1, unit);
+  if (bias_init_zero) {
+    bias.setZero();
+  } else {
+    bias.setRandUniform(-0.5, 0.5);
+  }
+
+  TensorDim dim = output_dim;
+  dim.height(input_dim.width());
+  dim.batch(1);
+  Tensor weight = initializeWeight(dim, weight_ini_type, status);
+  NN_RETURN_STATUS();
+
+  setParamSize(2);
+  paramsAt(0) = {std::move(weight), Tensor(weight.getDim()), "FC:weight"};
+  paramsAt(1) = {std::move(bias), Tensor(bias.getDim()), "FC:bias"};
+
+  TensorParam w =
+    TensorParam(paramsAt(0).weight, Tensor(weight.getDim()), "FC:weight", true);
```

Comment:
may be ```TensorParam(paramsAt(0).weight, paramsAt(0).grad, "FC:weight"}``` and TensorParam(paramsAt(1).weight, paramsAt(1).grad,"FC:bias",true)```

Reply:
yes, that is better. Will be updated to this style.